pyproject.toml

```diff
@@ -7,43 +7,133 @@ license = "MIT"
 readme = "README.rst"
 repository = "https://github.com/useblocks/sphinx-performance"
 documentation = "https://sphinx-performance.readthedocs.io/en/latest/"
-classifiers=[
-        'Development Status :: 5 - Production/Stable',
-        'Intended Audience :: Developers',
-        'License :: OSI Approved :: MIT License',
-        'Operating System :: OS Independent',
-        'Programming Language :: Python',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.7',
-        'Programming Language :: Python :: 3.8',
-        'Programming Language :: Python :: 3.9',
-        'Programming Language :: Python :: 3.10',
-        'Programming Language :: Python :: 3.11',
-        'Topic :: Documentation',
-        'Topic :: Utilities',
-        'Framework :: Sphinx',
-    ]
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
+    'Intended Audience :: Developers',
+    'License :: OSI Approved :: MIT License',
+    'Operating System :: OS Independent',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Topic :: Documentation',
+    'Topic :: Utilities',
+    'Framework :: Sphinx',
+]
 
 [tool.poetry.dependencies]
-python = "^3.7.0"
+python = ">=3.8,<4"
 click = "^8.0.3"
-rich = "^11.2.0"
 Jinja2 = "^3.0.3"
 memray = "^1.3.1"
-sphinx = "^5.3.0"
 pyinstrument = "^4.3.0"
+rich = "^11.2.0"
 snakeviz = "^2.1.1"
+sphinx = "^5.3.0"
 
-[tool.poetry.dev-dependencies]
+[tool.poetry.group.dev.dependencies]
+black = "^23.7.0"
+ruff = "^0.0.285"
+
+[tool.poetry.group.docs.dependencies]
+# some shared dependencies as the app itself is called in the docs build
```

Comment:
Do you really need duplicate the dependencies then? I normally only add the "extra" dependencies

Reply:
You're right, this will not work with `.readthedocs.yml` as it relies on pip.
I added an extra `docs` in which I just put `sphinxcontrib-programoutput`.

Reply:
BTW, the extra solution for docs is also not ideal, as docs requires not all deps of the app itself. This cannot be described in the current solution.